notepad.py

```diff
@@ -56,11 +58,17 @@ def handleCommand(message, command, uID):
         if len(words) < 2:
             message.room.send_message('Missing duration argument.')
             return
-        try:
-            time = float(words[1])
-        except:
-            message.room.send_message('Number expected as first argument, got %s.'%words[1])
+
+        pattern = '(?:(?P<days>\d+)d)? \s* (?:(?P<hours>\d+)h)? \s* (?:(?P<minutes>\d+)m)? \s* (?:(?P<seconds>\d+)s)?'
+        res = re.match(pattern, words[1])
```

Comment:
I'm pretty sure you need to pass `re.VERBOSE` in order to compensate for the whitespace I put between the groups.

Reply:
Ah right, forgot that. It was even in you intitial suggestion. Thanks.